src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -449,6 +490,12 @@ public static ApplicationModel.IResourceBuilder<T> WithExternalHttpEndpoints<T>(
         public static ApplicationModel.IResourceBuilder<T> WithHealthCheck<T>(this ApplicationModel.IResourceBuilder<T> builder, string key)
             where T : ApplicationModel.IResource { throw null; }
 
+        public static ApplicationModel.IResourceBuilder<TResource> WithHttpCommand<TResource>(this ApplicationModel.IResourceBuilder<TResource> builder, string path, string displayName, System.Func<ApplicationModel.EndpointReference>? endpointSelector, System.Net.Http.HttpMethod? method = null, string? httpClientName = null, System.Func<ApplicationModel.HttpCommandRequestContext, System.Threading.Tasks.Task>? configureRequest = null, System.Func<ApplicationModel.HttpCommandResultContext, System.Threading.Tasks.Task<ApplicationModel.ExecuteCommandResult>>? getCommandResult = null, string? commandName = null, System.Func<ApplicationModel.UpdateCommandStateContext, ApplicationModel.ResourceCommandState>? updateState = null, string? displayDescription = null, string? confirmationMessage = null, string? iconName = null, ApplicationModel.IconVariant? iconVariant = null, bool isHighlighted = false)
+            where TResource : ApplicationModel.IResourceWithEndpoints { throw null; }
+
+        public static ApplicationModel.IResourceBuilder<TResource> WithHttpCommand<TResource>(this ApplicationModel.IResourceBuilder<TResource> builder, string path, string displayName, string? endpointName = null, System.Net.Http.HttpMethod? method = null, string? httpClientName = null, System.Func<ApplicationModel.HttpCommandRequestContext, System.Threading.Tasks.Task>? configureRequest = null, System.Func<ApplicationModel.HttpCommandResultContext, System.Threading.Tasks.Task<ApplicationModel.ExecuteCommandResult>>? getCommandResult = null, string? commandName = null, System.Func<ApplicationModel.UpdateCommandStateContext, ApplicationModel.ResourceCommandState>? updateState = null, string? displayDescription = null, string? confirmationMessage = null, string? iconName = null, ApplicationModel.IconVariant? iconVariant = null, bool isHighlighted = false)
+            where TResource : ApplicationModel.IResourceWithEndpoints { throw null; }
```

Comment:
I'm of the opinion that these APIs (both `WithHttpCommand` overloads) have way too many parameters. It might be cleaner to encapsulate (or logically group) some of these parameters into a specific `DashboardOptions` object, or parameter object that group things together to help reduce the number of parameters. @davidfowl

Reply:
@DamianEdwards can you do the needful

Reply:
These expose the same parameters that the `WithCommand` methods do, plus the few extra specific to HTTP commands. I'm really not sure any other approach is better. Having this API take a new options object that isn't supported by `WithCommand` doesn't seem to make much sense. Are we thinking we introduce a new type like `CommandOptions` and use it for both `WithCommand` and `WithHttpCommand`, and mark the overloads of `WithCommand` that don't take it obsolete?

Reply:
This seems like a lesson that we need to keep learning ;)

Reply:
Yes, optional parameters at this scale are extremely hard to version. WithEndpoint is in the exact same jail 😄 .

Reply:
I’d be very happy if we stage a jailbreak on WithEndpoint in 10.0 so we can do proxyless persistent containers by default.

Reply:
So what's the consensus for 9.2? Change `WithCommand` and `WithHttpCommand` to use a new `CommandOptions` parameter and obsolete the existing `WithCommand` overload with all the optional parameters?

Reply:
I'd vote yes, move forward with `CommandOptions` parameter and obsolete the existing, but that's just me.

Reply:
#8276